Sources/SuperValidator/Validators/CreditCard.swift

```diff
@@ -0,0 +1,148 @@
+    /// Error Description for Credit Card Number
+    public enum CardNumberError: Error, LocalizedError {
+        case invalidCardNumber
+        public var errorDescription: String? {
+            switch self {
+            case .invalidCardNumber:
+                return nil
+            }
+        }
+    }
+    /// Error Description for CrediCard Expiry Date
+    public enum CardExpiryDateError: Error, LocalizedError {
+        case invalidExpiredDate
+        public var errorDescription: String? {
+            switch self {
+            case .invalidExpiredDate:
+                return nil
+            }
+        }
+    }
+    /// Error Description for CrediCard CSC (Card Security Code)
+    public enum CardCSCError: Error, LocalizedError {
+        case invalidCSC
+        public var errorDescription: String? {
+            switch self {
+            case .invalidCSC:
+                return nil
+            }
+        }
+    }
+}
+
+extension SuperValidator {
+    internal func creditCardNumberValidator(_ cardNumber: String,options: Option.CreditCard = .init()) -> Result<Void, CardNumberError> {
+        var tempCardNumber = cardNumber
+        /// Using separator
+        if options.separator.isNotEmpty {
+           tempCardNumber = ""
+            for number in cardNumber.components(separatedBy: options.separator) {
+                tempCardNumber += number
+            }
+        }
+        switch options.cardType {
+            case .visa:
+                guard tempCardNumber.matches(Regex.visa) else {
+                    return .failure(.invalidCardNumber)
+                }
+            case .mastercard:
+                guard tempCardNumber.matches(Regex.masterCard) else {
+                    return .failure(.invalidCardNumber)
+                }
+            case .amex:
+                guard tempCardNumber.matches(Regex.amex) else {
+                    return .failure(.invalidCardNumber)
+                }
+            case .discover:
+                guard tempCardNumber.matches(Regex.discover) else {
+                    return .failure(.invalidCardNumber)
+                }
+        }
+        return .success(())
+    }
+    
+    internal func creditCardExpiredDateValidator(expiryDate: String, options: Option.CreditCard = .init()) -> Result<Void, CardExpiryDateError> {
+        /// Expiry date should less than today
+        if expiryDate.toDate() ?? Date() < Date() {
+            return .failure(.invalidExpiredDate)
+        }
+        return .success(())
+    }
+    
+    internal func creditCardCSCValidator(csc: String, options: Option.CreditCard = .init()) -> Result<Void, CardCSCError> {
+        /// csc length should equal than card type
+        if csc.count != options.cardType.cardSecurityCodeLength {
+            return .failure(.invalidCSC)
+        }
+        return .success(())
+    }
+}
+
+extension String {
+    internal func toDate(withFormat format: String = "MM/yyyy") -> Date? {
```

Comment:
I think this should be fileprivate or make the name more straightforward like `toCreditCardExpiryDate` maybe?

Reply:
Okay, I'll change to straight forward